managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -777,19 +747,7 @@ public void asyncAddEntry(final byte[] data, int numberOfMessages, int offset, i
 
     @Override
     public void asyncAddEntry(ByteBuf buffer, AddEntryCallback callback, Object ctx) {
-        if (log.isDebugEnabled()) {
-            log.debug("[{}] asyncAddEntry size={} state={}", name, buffer.readableBytes(), state);
-        }
-
-        // retain buffer in this thread
-        buffer.retain();
-
-        // Jump to specific thread to avoid contention from writers writing from different threads
-        executor.execute(() -> {
-            OpAddEntry addOperation = OpAddEntry.createNoRetainBuffer(this, buffer, callback, ctx,
-                    currentLedgerTimeoutTriggered);
-            internalAsyncAddEntry(addOperation);
-        });
+        asyncAddEntry(buffer, 1, callback, ctx);
```

Comment:
Can we use -1 to keep original behaviour?

If this is set to 1, then the content originally written via this interface will go through the `broker metadata interceptor`.

For example, `commitTxn` or `abortTxn`, which will cause the index maintained in `AppendIndexMetadataInterceptor` to increment `incorrectly`.

Maybe the implicit behavior of this method(`asyncAddEntry(ByteBuf buffer, AddEntryCallback callback, Object ctx)`) is that calling this interface does not tuch the `broker metadata interceptor`.

Then, modifying the append of a normal transaction message should call the interface that passes in numberOfMessage.

Reply:
@shibd Please report an issue to track this. It seems that there are untested and undocumented behaviors in the original implementation that aren't tested if this change was possible.